src/BookAnalyzer.java

```diff
@@ -0,0 +1,73 @@
+import java.io.*;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.stream.*;
+import java.util.function.Function;
+
+public class BookAnalyzer {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        // Input the book name from the user
+        System.out.println("Enter a book name:");
+        String bookName = scanner.nextLine();
+
+        // Check if the book exists
+        File bookFile = new File(bookName + ".txt");
+        if (!bookFile.exists()) {
+            System.out.println("The book couldn't be found.");
+            return;
+        }
+
+        // Read the book and analyze text
+	int numOfWords = 0;
+	List<Map.Entry<String, Long>> sortedWordList;
+
+        try (Stream<String> lines = Files.lines(bookFile.toPath())) {
+	    List<String> lineList = lines.flatMap(line -> Arrays.stream(line.toLowerCase().replaceAll("[^a-z\\s]", "").split("\\s+"))).collect(Collectors.toList());
+	    numOfWords = new HashSet<String>(lineList).size();
+            sortedWordList = lineList
+		    .stream()
+		    .filter(word -> word.length() > 2)
+                    .collect(Collectors.collectingAndThen(
+                        Collectors.groupingBy(
+                            s -> s,
+                            Collectors.counting()
+                        ),
+                        map -> {
+                            List<Map.Entry<String, Long>> entryList = new ArrayList<>(map.entrySet());
+                            entryList.sort(Map.Entry.<String, Long>comparingByValue().reversed());
+                            return entryList;
+                        }
+    		    ));
```

Comment:
from my point of view it would be better to split this to more simple parts - Map<String, Long> wordToOccurence 
having this you get unique words (keys) and you can easily find the most popular words